pkg/services/mssqlfg/common_provision.go

```diff
@@ -0,0 +1,257 @@
+package mssqlfg
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	sqlSDK "github.com/Azure/azure-sdk-for-go/services/sql/mgmt/2017-03-01-preview/sql" // nolint: lll
+	"github.com/Azure/open-service-broker-azure/pkg/azure/arm"
+	"github.com/Azure/open-service-broker-azure/pkg/service"
+)
+
+func getServer(
+	ctx context.Context,
+	serversClient *sqlSDK.ServersClient,
+	resourceGroup string,
+	serverName string,
+	expectedVersion string,
+	expectedLocation string,
+) (string, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	result, err := serversClient.Get(
+		ctx,
+		resourceGroup,
+		serverName,
+	)
+	if err != nil {
+		return "", fmt.Errorf("error getting the sql server: %s", err)
+	}
+	if result.Name == nil {
+		return "", fmt.Errorf(
+			"can't find sql server %s in the resource group %s",
+			serverName,
+			resourceGroup,
+		)
+	}
+	if *result.Version != expectedVersion {
+		return "", fmt.Errorf(
+			"sql server version validation failed, "+
+				"expected version: %s, actual version: %s",
+			expectedVersion,
+			result.Version,
+		)
+	}
+	expectedLocation = strings.Replace(
+		strings.ToLower(expectedLocation),
+		" ",
+		"",
+		-1,
+	)
+	actualLocation := strings.Replace(
+		strings.ToLower(*result.Location),
+		" ",
+		"",
+		-1,
+	)
+	if expectedLocation != actualLocation {
+		return "", fmt.Errorf(
+			"sql server location validation failed, "+
+				"expected location: %s, actual location: %s",
+			expectedLocation,
+			actualLocation,
+		)
+	}
+	if *result.FullyQualifiedDomainName == "" {
+		return "", fmt.Errorf(
+			"sql server details doesn't contain FQDN",
+		)
+	}
+	return *result.FullyQualifiedDomainName, nil
+}
+
+func getDatabase(
```

Comment:
Actually, this function is "validate" instead of "get". When we say "getSomething", "something" should be one of the return values.

Reply:
The same below. But seems changing the naming convention will cause a log changes of the code...

Reply:
Good point. Will rename them.

Reply:
Inside the function, it is "get" as it calls the `get` method from SDK.